cmd/emit.go

```diff
@@ -0,0 +1,64 @@
+package cmd
+
+import (
+	"github.com/boson-project/func/cloudevents"
+	"github.com/google/uuid"
+	"github.com/ory/viper"
+	"github.com/spf13/cobra"
+)
+
+func init() {
+	e := cloudevents.NewEmitter()
+	root.AddCommand(emitCmd)
+	emitCmd.Flags().BoolP("local", "l", false, "Send the CloudEvent to the function running on the local host")
+	emitCmd.Flags().StringP("source", "s", e.Source, "CloudEvent source")
+	emitCmd.Flags().StringP("type", "t", e.Type, "CloudEvent type")
+	emitCmd.Flags().StringP("id", "", uuid.NewString(), "CloudEvent ID")
+	emitCmd.Flags().StringP("data", "", "", "Path to a local file containing CloudEvent data to be sent")
+}
+
+var emitCmd = &cobra.Command{
+	Use:   "emit",
+	Short: "Emit a CloudEvent to a function endpoint",
+	Long: `Emit event
+
+Emits a CloudEvent, sending it to the deployed function.
+`,
+	Example: `
+# Send a CloudEvent to the deployed function with no data and default values
+# for source, type and ID
+kn func emit
```

Comment:
This right here is the big benefit of bringing this into the func client library.  We can assume defaults that take into account the current Function context.  If we can keep it this simple, this is definitely going to be a win for Function authors.